backend/DB/api/routes.py

```diff
@@ -0,0 +1,239 @@
+
+
+@app.route('/api/pair/at_date/update/<date>', methods=['PUT'])
+def update_pair(date):
+    r = request.get_json()
+
+    if r is None or 'person1' not in r or 'person2' not in r:
+        abort(400)
+
+    pair = [Pair(r['person1'], r['person2'], date)]
+    queries.update_pair(pair[0])
+    return jsonify(format_pairs(pair)[0])
+
+
+def format_rewards(rewards):
+    output = []
+    for reward in rewards:
+        output.append({'reward_type': reward.reward_type, 'date': reward.date, 'used_reward': reward.used_reward})
+    return jsonify(output)
+
+
+@app.route('/api/reward/add', methods=['POST'])
+def add_reward():
+    r = request.get_json()
+
+    if r is None or 'reward_type' not in r:
+        abort(400)
+
+    if 'date' not in r:
+        reward = Reward(r['reward_type'].lower())
+    else:
+        reward = Reward(r['reward_type'].lower(), r['date'])
+    if 'used_reward' in r:
+        reward.used_reward = r['used_reward']
+    queries.add_reward(reward)
+    return format_rewards([reward])
+
+
+@app.route('/api/reward/all', methods=['GET'])
+def get_all_rewards():
+    rewards = queries.get_rewards()
+    return format_rewards(rewards)
+
+
+@app.route('/api/reward/unused/<reward_type>', methods=['GET'])
+def get_unused_reward_count(reward_type):
+    count = queries.get_unused_rewards_count_by_type(reward_type.lower())
+    return jsonify(count)
+
+
+@app.route('/api/reward/unused/earliest/<reward_type>', methods=['GET'])
+def get_earliest_unused_reward(reward_type):
+    return format_rewards([queries.get_earliest_unused_reward(reward_type)])
+
+
+@app.route('/api/reward/use/<reward_type>', methods=['PUT'])
+def use_reward(reward_type):
+    return format_rewards([queries.use_reward(reward_type)])
+
+
+@app.route('/api/threshold/add', methods=['POST'])
+def add_threshold():
+    r = request.get_json()
+
+    if r is None or 'reward_type' not in r or 'threshold' not in r:
+        abort(400)
+
+    threshold = Threshold(r['reward_type'], r['threshold'])
+    queries.add_threshold(threshold)
+    return jsonify([{'reward_type': threshold.reward_type, 'threshold': threshold.threshold}])
+
+
+@app.route('/api/threshold/get/<reward_type>', methods=['GET'])
+def get_threshold(reward_type):
+    threshold = queries.get_threshold(reward_type)
+    if threshold is None:
+        return jsonify([])
+    return jsonify([{'reward_type': threshold.reward_type, 'threshold': threshold.threshold}])
+
+
+@app.route('/api/threshold/update/<reward_type>', methods=['PUT'])
+def update_threshold(reward_type):
+    r = request.get_json()
+
+    if r is None or 'threshold' not in r:
+        abort(400)
+
+    threshold = queries.get_threshold(reward_type)
+    threshold.threshold = r['threshold']
+    queries.update_threshold(threshold)
+    return jsonify({'reward_type': threshold.reward_type, 'threshold': threshold.threshold})
+
+
+if __name__ == '__main__':
+    app.run(host='127.0.0.1', port=app.config.get("PORT", 5000))
```

Comment:
Since this is also a standard web app, i think it should just run on port 80

Reply:
Temporary due to proxy restrictions.